[Backport][ipa-4-6] ipatests: filter_users should be applied correctly if SSSD starts offline

amore17 opened the pull request:
This is manual back-port of : https://github.com/freeipa/freeipa/pull/3349

flo-renaud commented:
Hi @amore17 
pylint failed with
```
************* Module ipatests.test_integration.test_sssd

ipatests/test_integration/test_sssd.py:16: [E0611(no-name-in-module), ] No name 'osinfo' in module 'ipaplatform')

ipatests/test_integration/test_sssd.py:16: [E0401(import-error), ] Unable to import 'ipaplatform.osinfo')
```
You will need to adapt the PR because osinfo was introduced after ipa-4-6.

tiran commented:
It's probably not a good idea to backport 
1dadcd64b8400321fa9a407b96dece35046cbdf4 just for some tests. Also F28 is no longer supported. I propose that you remove the xfail for Fedora 28 instead.

amore17 commented:
@flo-renaud As per suggestion of @tiran I have removed back-port of https://github.com/freeipa/freeipa/commit/1dadcd64b8400321fa9a407b96dece35046cbdf4 and added xfail in another commit as sssd bulids are not available in fedora-27

flo-renaud commented on ipatests/test_integration/test_sssd.py:
```diff
@@ -0,0 +1,148 @@
+            tasks.clear_sssd_cache(self.master)
+
+    @pytest.mark.xfail(
+        reason='https://pagure.io/SSSD/sssd/issue/3978', strict=True)
```
I would replace with:
```
import ipaplatform
@pytest.mark.xfail(ipaplatform.NAME == 'fedora', reason='https://pagure.io/SSSD/sssd/issue/3978', strict=True)
```
Because the test is going to fail on fedora 27 but not on RHEL 7.8 (the tests for 7.8 are based on this branch ipa-4-6).

amore17 commented:
> Hi @amore17
> 
>     * the nightly definition needs to be updated in order to execute this test (ipatests/prci_definitions/nightly_ipa-4-6.yaml)
> 
>     * The backport also takes the tests from #3602 but this was not planned for ipa-4-6, are you sure you need the whole file?
Adding postponed label.
As there is test which is part of https://github.com/freeipa/freeipa/pull/3602 which is not back-ported yet to ipa-4-6 
Pending review on @wladich

flo-renaud commented:
Hi @amore17 
the backport looks ok but you should add a temp commit to launch the test in PRCI and check that it is properly marked as xfail.

flo-renaud commented:
Hi @amore17 
thanks for the update. The test was properly marked as xfailed, you can remove the temp commit.

flo-renaud commented:
Hi @amore17 
thanks for the update, tests are green, ACK.

fcami commented:
ipa-4-6:

* 4b70132c83f417b83aa4905de73f720336a90128 ipatests: filter_users should be applied correctly.
* d3b740e3df70c37bb3b7aa1fcd77acf5d68dc2bc Mark xfail for test_is_user_filtered
